lib/api/utilities/functions.php

```diff
@@ -81,172 +81,173 @@ function beans_remove_dir( $dir_path ) {
  *
- * @since 1.0.0
+ * @since 1.5.0
  *
- * @param string $url Url to be converted. Accepts only internal urls.
+ * @param string $url           Url to be converted. Accepts only internal urls.
+ * @param bool   $force_rebuild (Optional) Forces the rebuild of the root url and path.
  *
  * @return string Absolute path.
  */
-function beans_url_to_path( $url ) {
-
-	static $root, $blogdetails;
+function beans_url_to_path( $url, $force_rebuild = false ) {
+	static $root_path, $blogdetails;
+	$site_url = site_url();
+	if ( true === $force_rebuild ) {
+		$root_path   = '';
+		$blogdetails = '';
+	}
 
-	// Stop here if it is not an internal url.
-	if ( false === stripos( $url, parse_url( site_url(), PHP_URL_HOST ) ) ) {
-		return beans_sanitize_path( $url );
+	// Fix protocol. It isn't needed to set SSL as it is only used to parse the URL.
+	// Issue #63. Fixed.
+	if ( ! parse_url( $url, PHP_URL_SCHEME ) ) {
+		$original_url = $url;
+		$url          = 'http://' . ltrim( $url, '/' );
 	}
 
-	// Fix protocole. It isn't needed to set SSL as it is only used to parse the URL.
-	if ( preg_match( '#^(\/\/)#', $url ) ) {
-		$url = 'http:' . $url;
+	// It's not an internal URL. Bail out.
+	// Issue #64. #65. Fixed.
+	if ( false === stripos( parse_url( $url, PHP_URL_HOST ), parse_url( $site_url, PHP_URL_HOST ) ) ) {
+		return isset( $original_url ) ? $original_url : $url;
 	}
 
 	// Parse url and standardize backslashes.
-	$url = parse_url( $url, PHP_URL_PATH );
+	$url  = parse_url( $url, PHP_URL_PATH );
 	$path = wp_normalize_path( $url );
-	$explode = beans_get( 0, explode( '/' , trailingslashit( ltrim( $path, '/' ) ) ) );
 
-	// Maybe remove tilde from path.
-	if ( false !== stripos( $explode, '~' ) ) {
-		$path = preg_replace( '#\~[^/]*\/#', '', $path );
+	// Maybe remove tilde from path. -> Issue #66 - Fixed.
+	$trimmed_path = trailingslashit( ltrim( $path, '/' ) );
+	$maybe_tilde  = beans_get( 0, explode( '/', $trimmed_path ) );
+	if ( false !== stripos( $maybe_tilde, '~' ) ) {
+		$ends_with_slash = substr( $path, - 1 ) === '/';
+		$path            = preg_replace( '#\~[^/]*\/#', '', $trimmed_path );
+		if ( $path && ! $ends_with_slash ) {
+			$path = rtrim( $path, '/' );
+		}
 	}
 
 	// Set root if it isn't cached yet.
-	if ( ! $root ) {
-
+	if ( ! $root_path ) {
 		// Standardize backslashes and remove windows drive for local installs.
-		$root = wp_normalize_path( untrailingslashit( ABSPATH ) );
-		$set_root = true;
-
+		$root_path = wp_normalize_path( untrailingslashit( ABSPATH ) );
+		$set_root  = true;
 	}
 
-	// Remove subfolder if necessary.
-	if ( '' !== ( $subfolder = parse_url( site_url(), PHP_URL_PATH ) ) ) {
-
-		// Set root if it isn't cached.
-		if ( isset( $set_root ) ) {
-
-			// Remove subfolder.
-			$root = preg_replace( '#' . untrailingslashit( preg_quote( $subfolder ) ) . '$#', '', $root );
+	// If the subfolder exists for the root URL, then strip it off of the root path.
+	// Why? We don't want a double subfolder in the final path.
```

Comment:
This doesn't comply to the WP standards multi-line doc (see [here](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#5-2-multi-line-comments)).